src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -233,11 +249,17 @@ def _process_message(wrapper: Dict) -> None:
                         monitor_environment.last_checkin
                     )
 
+                # If the UUID is unset (zero value) generate a new UUID
+                if check_in_id.int == 0:
+                    guid = uuid.uuid4()
+                else:
+                    guid = check_in_id
+
                 check_in = MonitorCheckIn.objects.create(
                     project_id=project_id,
                     monitor=monitor,
                     monitor_environment=monitor_environment,
-                    guid=params["check_in_id"],
+                    guid=guid,
```

Comment:
just to be clear, previously this was casting the string to a UUID, now we're just providing it the object

Reply:
or was this object already a UUID that we just re-cast?

Reply:
it's no problem, the column is a UUID type anyway, it was actually being cast back to a UUID if this was a string